src/BloomBrowserUI/bookEdit/js/textOverPicture.ts

```diff
@@ -31,16 +40,237 @@ class TextOverPictureManager {
+                );
+            } else {
+                theOneTextOverPictureManager.setActiveElement(undefined);
+            }
+        }
+    }
+
+    public getActiveElement() {
+        return this.activeElement;
+    }
+
+    private setActiveElement(element: HTMLElement | undefined) {
+        if (element == undefined) {
+            this.activeElement = undefined;
+            return;
+        }
+
+        this.activeElement = element;
+        if (this.notifyBubbleChange) {
+            this.notifyBubbleChange(this.getSelectedItemBubbleSpec());
+        }
+    }
+
+    public turnOffBubbleEditing(): void {
+        if (this.isCalloutEditingOn === false) {
+            return; // Already off. No work needs to be done.
+        }
+        this.isCalloutEditingOn = false;
+
+        // TODO: Not sure if we want to add this or if it has an effect etc.
+        // Maybe we want to clear it?
+        // Maybe we'd be able to remember what was the previously selected active element if we don't get rid of it? That might be kinda nice too
+        //this.setActiveElement(undefined);
+
+        const canvas = document.getElementsByClassName(
+            "bubble-edit-generated"
+        )[0];
+        if (canvas && canvas.parentElement) {
+            Comical.convertCanvasToSvgImg(canvas.parentElement as HTMLElement);
+        }
+        Array.from(
+            document.getElementsByClassName("bloom-hideImageButtons")
+        ).forEach(e => e.classList.remove("bloom-hideImageButtons"));
+
+        // Clean up event listeners that we no longer need
+        Array.from(document.getElementsByClassName("bloom-editable")).forEach(
+            element => {
+                element.removeEventListener(
+                    "focus",
+                    TextOverPictureManager.onFocusSetActiveElement
+                );
+            }
+        );
+    }
+
+    public prepareToSavePage(): void {
+        // Review: do we need to call turnOffComicaling, or is that done in time anyway?
+    }
+
     public cleanUp(): void {
         WebSocketManager.closeSocket(kWebsocketContext);
     }
 
+    public getSelectedItemBubbleSpec(): BubbleSpec | undefined {
+        if (!this.activeElement) {
+            return undefined;
+        }
+        return Bubble.getBubbleSpec(this.activeElement);
+    }
+
+    public requestBubbleChangeNotification(
+        notifier: (bubble: BubbleSpec | undefined) => void
+    ): void {
+        this.notifyBubbleChange = notifier;
+    }
+
+    public detachBubbleChangeNotification(): void {
+        this.notifyBubbleChange = undefined;
+    }
+
+    public updateSelectedItemBubbleSpec(
+        newBubbleProps: BubbleSpecPattern
+    ): void {
+        if (!this.activeElement) {
+            return;
+        }
+
+        const activeBubble = new Bubble(this.activeElement);
+        activeBubble.mergeWithNewBubbleProps(newBubbleProps);
+        Comical.update(this.activeElement.parentElement!);
```

Comment:
Could we move this into Comical (as updateBubbleSpec(activeElement, newBubbleProps)? Eventually, there may be a more efficient way to achieve the same effect, especially with internal knowledge of Comical.